src/ccache.cpp

```diff
@@ -2526,8 +2538,8 @@ cc_process_args(struct args* args,
     }
 
     // These are always too hard.
-    if (compopt_too_hard(argv[i]) || str_startswith(argv[i], "-fdump-")
-        || str_startswith(argv[i], "-MJ")) {
+    if (compopt_too_hard(argv[i]) || str_startswith(argv[i], "-fdump-")) {
+      // -MJ + e.g. -MM (=TOO_HARD) - should still write out.ccmd?
```

Comment:
How "intrusive" (up to: what do to do in case of CCACHE_DISABLE), should this behaviour be? I think always writing the file instead of the compiler is a must to remain consistent. This would mean all the args have to be parsed before early bailout is possible.